unittests/Threading/Thread.cpp

```diff
@@ -0,0 +1,27 @@
+//===--- Thread.cpp - Tests swift::Thread functionality -------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Threading/Thread.h"
+#include "gtest/gtest.h"
+
+using namespace swift;
+
+// Check that Thread::stackBounds() returns a reasonable value or none at all.
+TEST(StackTest, stack_bounds) {
+  auto bounds = Thread::stackBounds();
+
+  if (bounds.has_value()) {
```

Comment:
Should this be `hasValue()` if it's an `llvm::Optional` rather than a `std::optional`?

Reply:
LLVM recently [deprecated](https://reviews.llvm.org/D131349) the camelCase Optional accessors in favor of the snake_case ones.  I'm open to using the deprecated ones if that's preferred.

Reply:
Oh, okay! I missed that.